### PR TITLE
fix(telemetry): better hostname parsing for telemetry COMPASS-8148

### PR DESCRIPTION
**Pull request by alenakhineika**
## Description

- Return hostname without port
- Fall back to original hostname from uri when resolving srv failed

### Checklist
- [x] New tests and/or benchmarks are included
- [ ] Documentation is changed or added
- [ ] I have signed the MongoDB Contributor License Agreement (https://www.mongodb.com/legal/contributor-agreement)

## Motivation and Context

- [x] Bugfix
- [ ] New feature
- [ ] Dependency update
- [ ] Misc

## Types of changes

- [ ] *Backport Needed*
- [x] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/compass/src/app/utils/telemetry.ts [hunk 1]
```diff
@@ -115,15 +115,15 @@ async function getHostnameForConnection(
...
+        looseValidation: true,
+      });
     }
-    connectionStringData = new ConnectionString(uri, {
-      looseValidation: true,
-    });
   }
 
-  return connectionStringData.hosts[0];
+  const [hostname] = (connectionStringData.hosts[0] ?? '').split(':');
```
Comment (addaleax) [inline packages/compass/src/app/utils/telemetry.ts:125]: I feel like this is code I've written elsewhere as well, but keep in mind that the hostname itself can also contain `:` if it is an IPv6 address

**Review by addaleax — APPROVED** [review]

File: packages/compass/src/app/utils/telemetry.ts (hunk 1 above)
Comment (alenakhineika) [reply packages/compass/src/app/utils/telemetry.ts:125]: Can it be Atlas host as well with an IPv6 address?

Comment (alenakhineika) [reply packages/compass/src/app/utils/telemetry.ts:125]: And another question, did we solve this in this other project or do we keep doing the same? I am wondering if we want to address it here or this is fine leave as is.

Comment (alenakhineika) [reply packages/compass/src/app/utils/telemetry.ts]: Note: As discussed in Slack we want to handle IPv6.

**Review by addaleax — APPROVED** [review]

**Pull request merged**